### PR TITLE
Add new Promisify Anything (AWS Athena edition) example

### PR DESCRIPTION
**Pull request by pcholakov**
This example shows how to use Restate to promisify a complex dependency, in this case executing queries on AWS Athena.
A long-polling HTTP client is set up to simulate long-polling against the Restate ingress endpoint that wraps a query's
result in a very simple promise. Using a consistent idempotency token turns this into a true durable promise.

### CODE REVIEW

File: typescript/promisify-anything/src/query-athena.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }),
+      );
+      return startQueryResult.QueryExecutionId as string;
+    } catch (error) {
+      if (error instanceof athena.AthenaServiceException)
+        if (error.$fault === "client") throw new restate.TerminalError(error.message);
+
+      throw error; // rethrow other exceptions -- side effect retry policy will handle them
+    }
+  });
```
Comment (pcholakov) [inline typescript/promisify-anything/src/query-athena.ts:29]: @tillrohrmann WDYT about this alternative:

```ts
const startQueryOutput = await ctx.sideEffect(
  stopOnError(
    (e) => (e as { $fault?: string }).$fault === "client",
    async () =>
      await client.send(
        new athena.StartQueryExecutionCommand({
          QueryString: query ?? QUERY_SQL,
          WorkGroup: "demo-workgroup",
          ClientRequestToken: requestId,
        }),
      ),
  ),
);
```

Comment (tillrohrmann) [reply typescript/promisify-anything/src/query-athena.ts:29]: I think this is a good improvement which makes things easier to understand :-)

Comment (pcholakov) [reply typescript/promisify-anything/src/query-athena.ts]: Done!

File: typescript/promisify-anything/src/query-athena.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            ClientRequestToken: requestId,
+          }),
+        ),
+    ),
+  );
+  const queryId = { QueryExecutionId: startQueryOutput.QueryExecutionId };
+
+  const state = await ctx.sideEffect(async () =>
+    assertFinalState((await client.send(new athena.GetQueryExecutionCommand(queryId))).QueryExecution?.Status?.State),
+  );
```
Comment (pcholakov) [inline typescript/promisify-anything/src/query-athena.ts:30]: @tillrohrmann what do you think about this - as an alternative to a for-loop or throwing inside the sideEffect body?

Comment (tillrohrmann) [reply typescript/promisify-anything/src/query-athena.ts:30]: Generally looks good. What might not be so easy to understand is that we are using Restate's retry functionality to wait for the query to complete. Maybe adding something like ". Retrying." to the message of the Error in `assertFinalState` can make this clearer?

**Comment by pcholakov** [comment]
Thanks for the suggestion – I made a dedicated "RetryableError" class, I
like that!

**Pull request closed without merging**